frontend/src/registration/actions.ts

```diff
@@ -72,96 +74,118 @@ export const clearErrorMessage = (
   });
 };
 
-export const performValidation = (
-  data: RegistrationModel,
-  nextStep?: string
-): OcimThunkAction<void> => async dispatch => {
-  // Placeholder for form validation method
-  dispatch({
-    type: Types.REGISTRATION_VALIDATION,
-    data
-  });
-  setTimeout(() => {
-    try {
-      // TODO
-      if (data.domain === 'existing') {
-        throw Error('Test error');
-      }
-      dispatch({ type: Types.REGISTRATION_VALIDATION_SUCCESS, data });
-      if (nextStep) {
-        dispatch(push(nextStep));
-      }
-      dispatch({ type: UIActionTypes.NAVIGATE_NEXT_PAGE });
-    } catch (e) {
-      const error = {
-        domain: 'Domain already exists!'
-      };
-      dispatch({
-        type: Types.REGISTRATION_VALIDATION_FAILURE,
-        error
-      });
-    }
-  }, 800);
-};
-
 export const performValidationAndStore = (
   data: RegistrationModel,
   nextStep?: string
 ): OcimThunkAction<void> => async dispatch => {
-  // Placeholder for form validation method
   dispatch({
     type: Types.REGISTRATION_VALIDATION,
     data
   });
-  setTimeout(() => {
-    try {
-      // TODO
-      if (data.domain === 'existing') {
-        throw Error('Test error');
-      }
+  const registrationData: any = { ...data };
+
+  V2Api.instancesOpenedxConfigValidate({
+    data: { ...registrationData }
+  })
+    .then(() => {
+      // If validation succeeds, then save data and go to next step
       dispatch({ type: Types.REGISTRATION_VALIDATION_SUCCESS, data });
       if (nextStep) {
         dispatch(push(nextStep));
       }
-      dispatch({ type: UIActionTypes.NAVIGATE_NEXT_PAGE });
-    } catch (e) {
-      const error = {
-        domain: 'Domain already exists!'
-      };
-      dispatch({
-        type: Types.REGISTRATION_VALIDATION_FAILURE,
-        error
+    })
+    .catch((e: any) => {
+      e.json().then((feedback: any) => {
+        // If validation fails, return error to form through state
+        const error = { ...feedback };
+        // Loop at each error message and join them.
+        // Also convert keys from snake_case to camelCase
+        Object.keys(error).forEach(key => {
+          error[toCamelCase(key)] = error[key].join();
+        });
```

Comment:
@xitij2000 The mechanism I use to link the feedback messages to the fields requires them to have the same name. 
It'd be good if we could make the autogenerated client to convert the values automatically, but since that would take a lot of time I just converted here.